projects/optic/src/commands/bundle/bundle.ts

```diff
@@ -0,0 +1,507 @@
+import { Command } from 'commander';
+import { ParseResult, getFileFromFsOrGit } from '../../utils/spec-loaders';
+import { OpticCliConfig } from '../../config';
+import {
+  OpenAPIV3,
+  sourcemapReader,
+  UserError,
+} from '@useoptic/openapi-utilities';
+import { isYaml, JsonSchemaSourcemap, writeYaml } from '@useoptic/openapi-io';
+import fs from 'node:fs/promises';
+import path from 'path';
+import yaml from 'yaml';
+
+import { errorHandler } from '../../error-handler';
+import { jsonPointerHelpers } from '@useoptic/json-pointer-helpers';
+import { Operation } from 'fast-json-patch';
+import * as jsonpatch from 'fast-json-patch';
+import sortby from 'lodash.sortby';
+const description = `dereference an OpenAPI specification`;
+
+const usage = () => `
+  optic bundle <file_path>
+  optic bundle <file_path> > dereference.yml
+`;
+const helpText = `
+Example usage:
+  $ optic bundle openapi-spec.yml > bundled.yml
+  `;
+
+export const registerBundle = (cli: Command, config: OpticCliConfig) => {
+  cli
+    .command('bundle')
+    .configureHelp({
+      commandUsage: usage,
+    })
+    .addHelpText('after', helpText)
+    .description(description)
+    .argument('[file_path]', 'openapi file to bundle')
+    .option('-o [output]', 'output file name')
+    .option(
+      '--filter-x-extensions [extensions]',
+      'extensions to filter when truthy value set'
+    )
+    .option(
+      '--include-x-extensions [extensions]',
+      'extensions to filter when truthy value set'
+    )
```

Comment:
I'm not convinced this is the right place for filtering. This feels like it should be another command that's run before and piped into this function, otherwise we're setting ourselves up to support `--filter-x-extensions` / `--include-x-extensions` on _every_ optic command

Reply:
Do you know how to hide options in commander. Talked to folks about hiding these and then moving them off